File: job-addon-api/src/main/java/com/frost_fox/jenkins/job_addon/addon/Addon.java
package com.frost_fox.jenkins.job_addon.addon;

import com.frost_fox.jenkins.job_addon.AddonContext;

import java.util.Objects;

public class Addon {

    private AddonContext context;
    private long estimation;

    public Addon(AddonContext context, long estimation) {
        this.context = context;
        this.estimation = estimation;
    }

    public String getName() {
        return context.getName();
    }

    public String getId() {
        return context.getId();
    }

    public long getEstimation() {
        return estimation;
    }

    @Override
    public boolean equals(Object addon) {
        return addon instanceof Addon && addon.hashCode() == hashCode();
    }

    @Override
    public int hashCode() {
        return Objects.hash(getName(), getId());
    }
}

File: job-addon-grid/src/main/frontend/src/job/Addon.js
export default class Addon {
    constructor(info) {
        this.name = info.name;
        this.id = info.id;
        this.estimation = info.estimation;
    }

    /**
     * @returns {string}
     */
    getName() {
        return this.name;
    }

    /**
     * @returns {string}
     */
    getId() {
        return this.id;
    }

    /**
     * @param comparable
     * @returns {boolean}
     */
    equals(comparable){
        return comparable instanceof Addon &&
            this.name === comparable.name &&
            this.id === comparable.id;
    }
}